### PR TITLE
PEP 621 project metadata

### PR DESCRIPTION
**Pull request by amyreese**
Moves all project metadata into `pyproject.toml`. Moves most tool configuration as well, but flake8 config goes into a `.flake8` file until they support toml config. Uses a `.flake8` config from other projects that ignores more of the lint rules that are covered by usort/black.

### CODE REVIEW

**Comment by jreese** [comment]
Looks like Python 3.6 support was dropped in setuptools 60.0.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [build-system]
-requires = ["setuptools >= 64", "setuptools-scm >= 6.4"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling", "hatch-vcs"]
```
Comment (thatch) [inline pyproject.toml:2]: Is this the recommended manner, as in they promise the api remains stable forever?

**Comment by thatch** [comment]
Reluctantly, fine.

File: pyproject.toml (hunk 1 above)
Comment (amyreese) [reply pyproject.toml:2]: I don't know if they're promising anything, but that's the recommended usage from the packaging guide.

Comment (ofek) [reply pyproject.toml:2]: Yes, the behavior of the existing build options will never change. For reproducible builds you'd want to pin though.

**Pull request merged**